### PR TITLE
Upgrade to yaml-cpp 0.7.0

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

**Comment by clalancette** [comment]
Rebased on the latest.  Here is another CI try:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15892/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10567/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13556/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16197/)
* Windows Debug [Build Status](http://ci.ros2.org/job/ci_windows/16198/)

File: yaml_cpp_vendor-extras.cmake.in [hunk 1]
```diff
@@ -5,7 +5,7 @@ if(NOT yaml-cpp_FOUND)
   if(WIN32)
     set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/yaml_cpp_vendor/CMake")
   else()
-    set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/yaml_cpp_vendor/lib/cmake/yaml-cpp")
+    set(yaml-cpp_DIR "${@PROJECT_NAME@_DIR}/../../../opt/yaml_cpp_vendor/share/cmake/yaml-cpp")
```
Comment (nuclearsandwich) [inline yaml_cpp_vendor-extras.cmake.in:8]: The commit message for this change is rather vague. I'm not sure why the change was made.

Comment (clalancette) [reply yaml_cpp_vendor-extras.cmake.in:8]: I cannot reproduce this yaml-cpp failure locally on macOS Monterey.  I tried both with and without `yaml-cpp` installed via brew, and in both cases all of the tests succeeded for me.  At this point, I can only assume that the failure was happening because of some interaction on the old macOS Mojave builders, but since we are no longer supporting that I don't think this should be held up on that.  I'm going to go ahead and make the changes as requested here, and then turn this into a full PR and run CI on it again.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16025/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10688/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16352/)
* Windows Debug [Build Status](http://ci.ros2.org/job/ci_windows/16353/)

**Comment by j-rivero** [comment]
There is one more `0.6` reference in https://github.com/ros2/yaml_cpp_vendor/pull/25/files#diff-b5b35010854e957d6bd0c2b492e798ff01b5265c02244afa4f09fc7665a1adf2L1. Not sure if that was leaved out in propose.

A part from that, changes looks good to me (Jammy warnings seems to me totally unrelated to the PR).

**Comment by clalancette** [comment]
> There is one more `0.6` reference in https://github.com/ros2/yaml_cpp_vendor/pull/25/files#diff-b5b35010854e957d6bd0c2b492e798ff01b5265c02244afa4f09fc7665a1adf2L1. Not sure if that was leaved out in propose.

Nope, I totally missed it.  Thanks for pointing it out!  I'll fix it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -71,7 +62,7 @@ endmacro()
 # NO_CMAKE_PACKAGE_REGISTRY used to avoid finding the library downloaded in WORKSPACE B
 # when building workspace A.
 # This should only find a system installed yaml-cpp and thus the environment hook isn't needed.
-find_package(yaml-cpp 0.6 QUIET NO_CMAKE_PACKAGE_REGISTRY)
+find_package(yaml-cpp QUIET NO_CMAKE_PACKAGE_REGISTRY)
```
Comment (wjwwood) [inline CMakeLists.txt:65]: Why was this removed but not updated? (just curious)

**Review by wjwwood — APPROVED** [review]
lgtm, just one question

File: CMakeLists.txt (hunk 2 above)
Comment (clalancette) [reply CMakeLists.txt:65]: Mostly because we don't do that in any of our other vendor packages.  Looking through all of the ones I see in ROS 2 right now, I can't find a single example where we specify the version when trying to find the system package.  So I thought I would be consistent here.

**Comment by clalancette** [comment]
Thanks for the reviews!  I'm going to go ahead and merge this one.

**Comment by Ace314159** [comment]
Will this be backported to foxy and galactic in order to fix #10 there as well?

**Comment by clalancette** [comment]
> Will this be backported to foxy and galactic in order to fix #10 there as well?

No, this is too big of a change for Foxy and Galactic.  If we really want to fix this on Foxy and Galactic, we should do something like #29.

**Comment by Ace314159** [comment]
I see, thanks for explaining! I made #30 and #31 for foxy and galactic.
